src/stream/src/executor/source/source_executor.rs

```diff
@@ -381,6 +307,10 @@ impl<S: StateStore> SourceExecutor<S> {
             target_state
         );
 
+        if source_desc.source.config.enable_mux_reader() {
+            stream.drop_data_stream();
+        }
+
```

Comment:
Why do we need this step? Isn't this included in `replace_data_stream`?

Reply:
Oh I see. So the main point is that we must drop the old data stream before calling `self.stream_reader_builder`. LGTM.